workflow/executor/resource_test.go

```diff
@@ -64,63 +64,65 @@ func TestResourceFlags(t *testing.T) {
 // TestResourcePatchFlags tests whether Resource Flags
 // are properly passed to `kubectl patch` command
 func TestResourcePatchFlags(t *testing.T) {
+	fakeFlags := []string{"pod", "mypod"}
 	fakeClientset := fake.NewSimpleClientset()
-	manifestPath := "../../examples/hello-world.yaml"
-	buff, err := os.ReadFile(manifestPath)
-	assert.NoError(t, err)
-	fakeFlags := []string{"kubectl", "patch", "--type", "strategic", "-p", string(buff), "-f", manifestPath, "-o", "json"}
-
 	mockRuntimeExecutor := mocks.ContainerRuntimeExecutor{}
 
-	template := wfv1.Template{
-		Resource: &wfv1.ResourceTemplate{
-			Action: "patch",
-			Flags:  fakeFlags,
+	tests := []struct {
+		name           string
+		patchType      string
+		appendFileFlag bool
+		manifestPath   string
+	}{
+		{
+			name:           "strategic -f --patch-file",
+			patchType:      "strategic",
+			appendFileFlag: true,
+			manifestPath:   "../../examples/hello-world.yaml", // any YAML with a `kind`
```

Comment:
I'd rather this (and the other `manifestPath`s under test here were specific to the test suite, not overloading other files which have their own purpose. The effect of changing or deleting one of these (albeit unlikely) and seeing this test failing would be very surprising.

Reply:
I agree, the existing tests already did that though, so I did not change that test behavior in this PR. Changing that seemed to require a more substantial refactor, as we don't have a `testdata` dir for unit tests (this requires a file to read in) or much use of full manifests in units, so I deferred that for the purposes of this fix PR.

I could make a separate PR for that, but we'd have to think of a strategy to apply more generically to these types of unit tests

Reply:
Ok, that's reasonable.